Fixes issue #127 (Bad Content-type returned)

GaryTowers opened the pull request:
This fixes https://github.com/strongloop/loopback-component-storage/issues/127

Fixed by reading `filename` using system's `path` and passing it to `res.type` to set the proper Content-type instead of passing the `file` param.

This solves potential issues with other providers like `AWS` when the `file` param contains slashes appended to the file name. For example: **file:** `avatars/user1.png` 

This fix passes the `Download` test which also tests for the `Content-type` so no need to rewrite it.

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

superkhau commented:
Can you add some tests to verify your changes and prevent regressions in the future?

superkhau commented on lib/storage-handler.js:
```diff
@@ -196,7 +197,8 @@ function handleError(res, err) {
 exports.download = function(provider, req, res, container, file, cb) {
-
+  
+  var filename = path.basename(file);
```
fileName

superkhau commented on lib/storage-handler.js:
```diff
@@ -247,7 +249,7 @@ exports.download = function(provider, req, res, container, file, cb) {
       var reader = provider.download(params);
 
-      res.type(file);
+      res.type(filename);
```
ditto

superkhau commented on lib/storage-handler.js:
```diff
@@ -234,7 +236,7 @@ exports.download = function(provider, req, res, container, file, cb) {
           var reader = provider.download(params);
 
-          res.type(file);
+          res.type(filename);
```
ditto

GaryTowers commented:
Done, corrected typo.

The test already exists and can be found at https://github.com/strongloop/loopback-component-storage/blob/master/test/upload-download.test.js#L245 The changes I added wont break the test so the PR is safe from regressions. You can run it with mocha `mocha test/upload-download.test.js`

superkhau commented:
@slnode test please

GaryTowers commented:
Woah, any idea what and why those tests are failing? @superkhau

superkhau commented:
The CI server ran out of space, I already asked @rmg about it and he is on it. We'll just wait until I hear back and rerun the tests.

superkhau commented:
@slnode test please

GaryTowers commented:
Awesome, thanks @superkhau

rmg commented:
@slnode test please

GaryTowers commented:
@rmg @superkhau any updates on this issue? is the server ready for running another test round?

jaime-franco commented:
@superkhau @GaryTowers, Any news on this?

rmg commented:
@jaime-franco @GaryTowers the builds are up to date. The only failure is a new linter that was added and is complaining about the commit message on https://github.com/strongloop/loopback-component-storage/pull/129/commits/161390d71256d69fe7e624e8c72257891cf5686e having too long of a summary line.

I don't know if all of our repos are actually trying to enforce it or not, or what the plan is for community contributions, but the linters have been configured for all strongloop repos as a starting point.

It is trying to enforce the standard 50/72 git commit formatting conventions (50 cols, blank line, 72 cols):
```
summary line that is 50 chars or less

More detailed description of change here, in 0 or more paragraphs,
wrapped at 72 characters.
```

Other than the one commit message, the PR appears to build correctly.

GaryTowers commented:
Awesome. Thanks @rmg. Does that mean I need to issue a new commit following the 50/72  rule so the PR is merged?

superkhau commented:
@GaryTowers Yes, we should be good to go after fixing the long summary.

GaryTowers commented:
Hi @superkhau I fixed changed the long summary to a short line. The commit linter passes but now I see other things failing on jenkins side. I tried opening the details but I cannot access them, could you double check?

superkhau commented:
@GaryTowers We can ignore those errors for now, there are still some issues with CI ATM. Please squash your commits and let's get it merged.

GaryTowers commented:
@superkhau commits squashed 👍 All ready to merge

superkhau commented:
LGTM. Thanks for the contribution! ;)

Amir-61 commented:
Hey @superkhau and @GaryTowers ,

I don't think if the commit message like `Fix for issue #127` is really appropriate; The commit message should tell the others know what the commit was about without finding the issue in GH,...

superkhau commented:
@Amir-61 Nice catch, I didn't really read the commit message. @GaryTowers Please provide more descriptive commit messages next time (ie. `Fix for bad content-type returned`). Anyways, it's a nitpick and I already merged it in. ;)

alemhnan commented:
@superkhau I've stumbled in this error. Has been done a release with this merge? I am using version 1.8.1 and doesn't have this change?

superkhau commented:
@alemhnan Sorry for the late reply, I just did a release for 1.9.0. Enjoy. :beer:
